eng/Packages.Data.props

```diff
@@ -192,6 +192,7 @@
   <ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Developer.MicrosoftPlaywrightTesting'))">
     <PackageReference Update="Microsoft.TestPlatform.ObjectModel" Version="17.10.0" />
     <PackageReference Update="NUnit" Version="3.13.2" />
+    <PackageReference Update="Microsoft.Extensions.Logging" Version="2.0.0" />
```

Comment:
This was added as per recommendation in APIView review to use Microsoft.Extensions.Logging's ILogger interface. Added the dependency under this condition since it's used only in Extension libraries.